Makefile

```diff
@@ -1,3 +1,3 @@
 all: src/*.cpp inc/*.hpp
-	g++ -std=c++17 src/*.cpp -o main
+	g++ -std=c++17 -Werror -Iinc src/*.cpp -o main
```

Comment:
You are missing some additional flags `-Wall -Wextra -pedantic`. They will make this code not compiling.